File: app/src/main/java/com/example/test_loggin/MainActivity.java
package com.example.test_loggin;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.example.test_loggin.data.Usuario;

public class MainActivity extends AppCompatActivity implements View.OnClickListener{
    EditText user, pass;
    Button btnIngresar, btnRegistrar;
    daoUsuarios dao;
    //Button btoReg; //declaro el boton

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        user=(EditText)findViewById(R.id.editTextTextPersonName);
        pass=(EditText)findViewById(R.id.editTextTextPassword);
        btnIngresar=(Button)findViewById(R.id.Ingresar);
        btnRegistrar=(Button)findViewById(R.id.ButtonReg);
        btnIngresar.setOnClickListener(this);
        btnRegistrar.setOnClickListener(this);
        dao=new daoUsuarios(this);

    }

    @Override
    public void onClick(View v) {
        switch (v.getId()){
            case R.id.Ingresar:
                String u=user.getText().toString();
                String p=pass.getText().toString();
                if (u.equals("")&& p.equals("")){
                    Toast.makeText(this,"Campos Vacios",Toast.LENGTH_LONG).show();
                }else if (dao.login(u, p) == 1){ //revisar validacion
                    Usuario ux= dao.getUsuario(u,p);
                    Toast.makeText(this,"LOGIN Exitoso",Toast.LENGTH_LONG).show();
                    Intent i2=new Intent(MainActivity.this,Menu.class);
                   i2.putExtra("id", ux.getId()); //paso ID del usuario logueado
                    startActivity(i2);
                }

                break;
            case R.id.ButtonReg:
                Intent i=new Intent(MainActivity.this,Registrarse.class);
                startActivity(i);
                break;
        }
    }
}

File: app/src/main/java/com/example/test_loggin/MainActivity.kt
package com.example.test_loggin

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.example.test_loggin.MainActivity

class MainActivity : AppCompatActivity(), View.OnClickListener {
    var user: EditText? = null
    var pass: EditText? = null
    var btnIngresar: Button? = null
    var btnRegistrar: Button? = null
    var dao: daoUsuarios? = null

    //Button btoReg; //declaro el boton
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        user = findViewById<View>(R.id.editTextTextPersonName) as EditText
        pass = findViewById<View>(R.id.editTextTextPassword) as EditText
        btnIngresar = findViewById<View>(R.id.Ingresar) as Button
        btnRegistrar = findViewById<View>(R.id.ButtonReg) as Button
        btnIngresar!!.setOnClickListener(this)
        btnRegistrar!!.setOnClickListener(this)
        dao = daoUsuarios(this)
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.Ingresar -> {
                val u = user!!.text.toString()
                val p = pass!!.text.toString()
                if (u == "" && p == "") {
                    Toast.makeText(this, "Campos Vacios", Toast.LENGTH_LONG).show()
                } else if (dao!!.login(u, p) == 1) { //revisar validacion
                    val ux = dao!!.getUsuario(u, p)
                    Toast.makeText(this, "LOGIN Exitoso", Toast.LENGTH_LONG).show()
                    val i2 = Intent(this@MainActivity, Menu::class.java)
                    i2.putExtra("id", ux.id) //paso ID del usuario logueado
                    startActivity(i2)
                }
            }
            R.id.ButtonReg -> {
                val i = Intent(this@MainActivity, Registrarse::class.java)
                startActivity(i)
            }
        }
    }
}